File: lab05/Lab/echon.pl
#!/usr/bin/perl -w

# determine if the correct number of arguments was given
if ($#ARGV+1 != 2){
	print "Usage: ./echon.pl <number of lines> <string> at ./echon.pl line 3.", "\n";
	exit(1);
}

$max = $ARGV[0];
$string = $ARGV[1];
# check to see if second arg is actually a number
if ($max =~ /[^0-9]/){
	print "Usage: ./echon.pl <number of lines> <string> at ./echon.pl line 3.", "\n";
	exit(1);
}

$i = 0;
while ($i < $max){
	print $string, "\n";
	$i = $i + 1;
}

File: lab06/echon.py
#!/usr/bin/python
import sys
import re

# Demonstrate traversing through arguments
# for arg in sys.argv:
#    print arg

argumentList = sys.argv
#print argumentList
#print len(argumentList)
# check if the user supplied the correct number of arguments
if len(argumentList) != 3:
    sys.exit("Usage: ./echon.py <number of repetitions> <string to repeat>")

maxIterations = argumentList[1];
string = argumentList[2];

# check to see if maxIterations is actually a number
isNumRegex = r"^[0-9]+$"
if not (re.search(isNumRegex, maxIterations)):
    sys.exit("./echon.py: Argument 1 must be a non-negative integer")

i = 0
while i < int(maxIterations):
    print string
    i += 1


# print "Max Iterations = %d and string to print is: %s" %( int(maxIterations), string )
